lib/gooddata/lcm/actions/collect_data_product.rb

```diff
@@ -31,18 +31,13 @@ def call(params)
           domain = client.domain(domain_name) || fail("Invalid domain name specified - #{domain_name}")
 
           if params.key?(:data_product)
-            begin
-              data_product = domain.data_products(params.data_product)
-            rescue RestClient::BadRequest
-              fail 'the specified DataProduct does not exist'
-            end
+            data_product_id = params.data_product
```

Comment:
Refers to 'params' more than self (maybe move it to another class?) (FeatureEnvy)